File: ios/NativeAlert.m
//
//  NativeAlert.m
//  myUnicorn
//
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTUtils.h> // using React's Utils to help us find Root View Controller
#import "NativeAlertBridgingHeader.h"

NSHashTable *_alertControllers;

@interface NativeAlert()

@end

@implementation NativeAlert
{
  
}

RCT_EXPORT_MODULE()

- (dispatch_queue_t)methodQueue
{
  return dispatch_get_main_queue();
}

- (void)invalidate
{
  for (UIAlertController *alertController in _alertControllers) {
    [alertController.presentingViewController dismissViewControllerAnimated:YES completion:nil];
  }
}


RCT_EXPORT_METHOD(openAlert:(NSString *)title
                  :(NSString *)msg)
{
  
  UIViewController *presentingController = RCTPresentedViewController();
  if (presentingController == nil) {
    return;
  }
  
  UIAlertController *alertController = [UIAlertController
                                        alertControllerWithTitle:title
                                        message:msg
                                        preferredStyle:UIAlertControllerStyleAlert];
  
  
  
  [alertController addAction:[UIAlertAction actionWithTitle:@"Ok"
                                                      style:UIAlertActionStyleDefault
                                                    handler:^(__unused UIAlertAction *action) {
                                                      // no actions needed
                                                    }]];

  if (!_alertControllers) {
    _alertControllers = [NSHashTable weakObjectsHashTable];
  }
  [_alertControllers addObject:alertController];
  
  [presentingController presentViewController:alertController animated:YES completion:nil];
}

@end

File: ios/NativeAlert.swift
//
//  NativeAlert.swift
//  myUnicorn
//
//  Not Currently in Use
//

import Foundation
import UIKit

@objc(NativeAlertSwift)
open class NativeAlertSwift: UIViewController {
  
  @objc open func openAlert(_ msg: String) {
    let alert = UIAlertController(title: "Alert", message: msg, preferredStyle: .alert)
    alert.addAction(UIAlertAction(title: "OK", style: .default))
    self.present(alert, animated: true, completion: nil) // using self.present states error: whose view is not in the window hierarchy
  }
  
}
